sys/include/net/ng_rpl.h

```diff
@@ -7,27 +8,413 @@
+#endif
+
+/**
+ * @brief   Default message queue size to use for the RPL thread.
+ */
+#ifndef NG_RPL_MSG_QUEUE_SIZE
+#define NG_RPL_MSG_QUEUE_SIZE  (8U)
+#endif
+
+/**
+ * @brief   Static initializer for the all-RPL-nodes multicast IPv6
+ *          address (ff02::1a)
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6">
+ *          RFC 6550, section 6
+ *      </a>
+ */
+#define NG_IPV6_ADDR_ALL_RPL_NODES {{ 0xff, 0x02, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x1a }}
+
+/**
+ * @brief   Message type for lifetime updates
+ */
+#define NG_RPL_MSG_TYPE_LIFETIME_UPDATE     (0x0900)
+
+/**
+ * @brief   Message type for trickle intervals
+ */
+#define NG_RPL_MSG_TYPE_TRICKLE_INTERVAL    (0x0901)
+
+/**
+ * @brief   Message type for trickle callbacks
+ */
+#define NG_RPL_MSG_TYPE_TRICKLE_CALLBACK    (0x0902)
+
+/**
+ * @brief   Message type for handling DAO sending
+ */
+#define NG_RPL_MSG_TYPE_DAO_HANDLE  (0x0903)
+
+/**
+ * @brief   Message type for handling DODAG cleanup
+ */
+#define NG_RPL_MSG_TYPE_CLEANUP_HANDLE  (0x0904)
+
+/**
+ * @brief   Infinite rank
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-17">
+ *          RFC 6550, section 17
+ *      </a>
+ */
+#define NG_RPL_INFINITE_RANK (0xFFFF)
+
+/**
+ * @brief   Default minimum hop rank increase
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-17">
+ *          RFC 6550, section 17
+ *      </a>
+ */
+#define NG_RPL_DEFAULT_MIN_HOP_RANK_INCREASE (256)
+
+/**
+ * @brief   Maximum rank increase
+ */
+#define NG_RPL_DEFAULT_MAX_RANK_INCREASE (0)
+
+/**
+ * @brief   Number of implemented Objective Functions
+ */
+#define NG_RPL_IMPLEMENTED_OFS_NUMOF (1)
+
+/**
+ * @brief   Default Objective Code Point (OF0)
+ */
+#define NG_RPL_DEFAULT_OCP (0)
+
+/**
+ * @name RPL Mode of Operations
+ * @{
+ */
+#define NG_RPL_MOP_NO_DOWNWARD_ROUTES  (0x00)
+#define NG_RPL_MOP_NON_STORING_MODE    (0x01)
+#define NG_RPL_MOP_STORING_MODE_NO_MC  (0x02)
+#define NG_RPL_MOP_STORING_MODE_MC     (0x03)
+/** default MOP set on compile time */
+#ifndef NG_RPL_DEFAULT_MOP
+#   define NG_RPL_DEFAULT_MOP NG_RPL_MOP_STORING_MODE_NO_MC
+#endif
+/** @} */
+
+/**
+ * @name Sequence counter handling
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-7.2">
+ *          Sequence Counter Operation
+ *      </a>
+ * @{
+ */
+#define NG_RPL_COUNTER_MAX                 (255)
+#define NG_RPL_COUNTER_LOWER_REGION        (127)
+#define NG_RPL_COUNTER_SEQ_WINDOW          (16)
+#define NG_RPL_COUNTER_INIT                (NG_RPL_COUNTER_MAX - NG_RPL_COUNTER_SEQ_WINDOW + 1)
+
+static inline uint8_t NG_RPL_COUNTER_INCREMENT(uint8_t counter)
+{
+    return ((counter > NG_RPL_COUNTER_LOWER_REGION) ?
+            ((counter == NG_RPL_COUNTER_MAX) ? counter = 0 : ++counter) :
+            ((counter == NG_RPL_COUNTER_LOWER_REGION) ? counter = 0 : ++counter));
+}
+
+static inline bool NG_RPL_COUNTER_IS_INIT(uint8_t counter)
+{
+    return (counter > NG_RPL_COUNTER_LOWER_REGION);
```

Comment:
I took that part from the old implementation. I propose to leave it as it is and do some cleanups with follow-up PRs. As far as I know this counter operation is described in [RFC1982](https://tools.ietf.org/html/rfc1982). I can imagine to work on a separate module for this lollipop like sequence number operation, because it is used in various protocols. Afterwards this code here can be thrown out.